tasks/section_4/cis_4.6.x.yml

```diff
@@ -89,17 +89,22 @@
 
 - name: "4.6.5 | PATCH | Ensure default user umask is 027 or more restrictive"
   block:
-      - name: "4.6.5 | PATCH | Ensure default user umask is 027 or more restrictive | Set umask for /etc/login.defs pam_umask settings"
+      - name: "4.6.5 | PATCH | Ensure default user umask is 027 or more restrictive | Set umask for /etc/login.defs and /etc/profile"
         ansible.builtin.lineinfile:
             path: "{{ item.path }}"
-            regexp: '(?i)(umask\s*)'
+            regexp: '(?i)(umask\s*\d\d\d)'
             line: '{{ item.line }} 027'
         with_items:
-            - { path: '/etc/bashrc', line: 'umask' }
             - { path: '/etc/profile', line: 'umask' }
             - { path: '/etc/login.defs', line: 'UMASK' }
 
-      - name: "4.6.5 | PATCH | Ensure default user umask is 027 or more restrictive | Set umask for /etc/bashrc"
+      - name: "4.6.5 | PATCH | Ensure default user umask is 027 or more restrictive | Set umask for /etc/bashrc" 
+        ansible.builtin.replace:
+            path: /etc/bashrc
+            regexp: '\s+umask\s*\d\d\d'
+            replace: '\numask 027'
```

Comment:
why add a new line ?
```suggestion
            replace: 'umask 027'
```

Reply:
As I presented in the issue responsible for this PR:
"Actual Behavior
 Firstly, [...]
 Secondly, the way the task is written, it does not edit the /etc/bashrc file in a proper manner, only modifying one of 
 the 
 umask lines, not all of them."

The only way it worked for my tests was to add 'umask 027' with \n in order to edit properly both umask lines from /etc/bashrc.

Reply:
I am sure this would get you where you want too.
```
ansible.builtin.replace:
            path: /etc/bashrc
            regexp: '^(\s+umask\s*)\d\d\d'
            replace: '\1 027'
```

Reply:
I tried you approach but unfortunately, it did not work. This is the content of the `/etc/bashrc` file after executing the task with `replace: '\1 027'`: 

![image](https://github.com/ansible-lockdown/AMAZON2023-CIS/assets/143085010/e93ad89c-dd2a-499b-ace4-953255e9d324)

Using `\numask 027` still works fine and modifies all the lines with umask.

Reply:
Does this work?
 
This should only change those not 027 or more restrictive and whether 3 or 4 characters

```
regexp: '^(?i)(\s*umask)\s+(?\!\d*[2,7]7)\d{3,4}'
replace: '\1 027'
```

Reply:
I'll try to test it!

Reply:
Strange i created a dummy file to test this 
the hashed items are how it looked to start and below is what changed.

Its working as expected there. 

```
##Original Settings
#UMASK 022
#  UMASK 0022

#umask 0002
#  umask 002

#UMASK 000
#  UMASK 0000

## These should not change

#umask 077
#  umask 0077

###########
## This should change to 027

UMASK 027
  UMASK 027

umask 027
  umask 027

UMASK 027
  UMASK 027

## These should not change

umask 077
  umask 0077
```

Reply:
Let me rebuild again and test on a new build too.

Reply:
Just ran manually on bashrc

```
grep -i umask /etc/bashrc
    # By default, we want umask to get set. This sets it for non-login shell.
       umask 002
       umask 022
[ec2-user@test ~]$ cp -p /etc/bashrc /etc/bashrc_orig
cp: cannot create regular file '/etc/bashrc_orig': Permission denied
[ec2-user@test ~]$ sudo cp -p /etc/bashrc /etc/bashrc_orig
[ec2-user@test ~]$ grep -i umask /etc/bashrc
    # By default, we want umask to get set. This sets it for non-login shell.
       umask 027
       umask 027
```

Just checking we're in the same place i'm using this regexp

```
ansible.builtin.replace:
    path: /etc/bashrc
    regexp: '^(?i)(\s*umask)\s+(?\!\d*[2,7]7)\d{3,4}'
    replace: '\1 027'
```

Reply:
I used the same regexp as well, still, it did not modify the lines...

Reply:
Ok lets close this PR i will add the fix to the new Feb24 ive been working on, there seems to be so many commits for one line change for this control. 
Let see if we reset and how we get on then, i have a feeling something is not quite right somewhere.

Reply:
I agree! Thank you for your time!

Reply:
Not a problem at all, will ensure credit added for the update 
thank you for the time